src/ecc_bitcoin.c

```diff
@@ -104,22 +105,28 @@ void libsecp256k1_ecc_context_destroy(void)
 
 
 int libsecp256k1_ecc_sign_digest(const uint8_t *private_key, const uint8_t *data,
-                                 uint8_t *sig, ecc_curve_id curve)
+                                 uint8_t *sig, uint8_t *recid, ecc_curve_id curve)
 {
     (void)(curve);
-    secp256k1_ecdsa_signature signature;
+    secp256k1_ecdsa_recoverable_signature signature;
 
     if (!libsecp256k1_ctx) {
         libsecp256k1_ecc_context_init();
     }
 
-    if (secp256k1_ecdsa_sign(libsecp256k1_ctx, &signature, (const unsigned char *)data,
-                             (const unsigned char *)private_key, secp256k1_nonce_function_rfc6979, NULL)) {
-        int i;
+    if (secp256k1_ecdsa_sign_recoverable(libsecp256k1_ctx, &signature,
+                                         (const unsigned char *)data,
+                                         (const unsigned char *)private_key, secp256k1_nonce_function_rfc6979, NULL)) {
+        int i, recid_ = 0xFF;
+        secp256k1_ecdsa_recoverable_signature_serialize_compact(libsecp256k1_ctx, sig,
+                &recid_, &signature);
         for (i = 0; i < 32; i++) {
             sig[i] = signature.data[32 - i - 1];
```

Comment:
Actually `secp256k1_ecdsa_recoverable_signature_serialize_compact` already fills `sig`. Not sure why the loop was included... `secp256k1_ecdsa_recoverable_signature_parse_compact` works in the other direction.
Updated/fixed in the latest (rebased) commit.